### PR TITLE
Feat/new archi

### PR DESCRIPTION
**Pull request by lucas-barbero**

### CODE REVIEW

File: RunningOrder/Sprint/NewSprintView.swift [hunk 1]
```diff
@@ -39,7 +39,9 @@ struct NewSprintView: View {
 
     private func createSprint() {
         guard areAllFieldsFilled else { return }
-        self.createdSprint = Sprint(number: number!, name: name, colorIdentifier: "holiday blue", stories: [])
+        let newSprint = Sprint(number: number!, name: name, colorIdentifier: "holiday blue")
+        createdSprint = newSprint
+        Store.data[newSprint] = []
```
Comment (zarghol) [inline RunningOrder/Sprint/NewSprintView.swift:44]: the store shouldn't be here, but at the storing of the sprint, in the super view / viewModel

File: RunningOrder/Sprint/SprintDetail.swift [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+//
+//  Created by Lucas Barbero on 05/08/2020.
+//  Copyright © 2020 Worldline. All rights reserved.
+//
+
+import SwiftUI
+
+struct SprintDetail: View {
+
+    var sprint: Sprint
```
Comment (zarghol) [inline RunningOrder/Sprint/SprintDetail.swift:13]: let ? ou binding ?

Comment (zarghol) [reply RunningOrder/Sprint/SprintDetail.swift:13]: since you have a viewModel, the sprint should be in it

File: RunningOrder/Sprint/SprintDetail.swift [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    init(sprint: Sprint) {
+        self.sprint = sprint
+        viewModel = SprintDetailViewModel(sprint: sprint)
+    }
+    var body: some View {
+
+            NavigationView {
+                StoryList(header: "Sprint \(sprint.number) - \(sprint.name)", stories: $viewModel.stories) // TODO Construct/Pass the head in a different way ?
+                    .onAppear {
```
Comment (zarghol) [inline RunningOrder/Sprint/SprintDetail.swift:25]: why another onAppear, you already have one

File: RunningOrder/MainView.swift [hunk 4]
```diff
@@ -13,7 +13,6 @@ struct MainView: View {
     var body: some View {
         NavigationView {
             SprintList()
-                .listStyle(SidebarListStyle())
```
Comment (zarghol) [inline RunningOrder/MainView.swift:16]: why ?

File: RunningOrder/Sprint/SprintDetail.swift (hunk 2 above)
Comment (zarghol) [reply RunningOrder/Sprint/SprintDetail.swift:13]: use viewModel.sprint everywhere x)

File: RunningOrder/Sprint/SprintList.swift [hunk 5]
```diff
@@ -51,8 +51,13 @@ struct SprintList: View {
             }
             .padding(.all, 8.0)
             .buttonStyle(PlainButtonStyle())
-        }.sheet(isPresented: $showNewSprintModal) {
-            NewSprintView(createdSprint: self.$sprints.appendedElement)
+        }
+        .listStyle(SidebarListStyle())
+        .sheet(isPresented: $showNewSprintModal) {
+            NewSprintView(createdSprint: $viewModel.sprints.appendedElement)
+        }
```
Comment (zarghol) [inline RunningOrder/Sprint/SprintList.swift:58]: why this change ?

File: RunningOrder/Story/NewStoryView.swift [hunk 6]
```diff
@@ -41,13 +43,18 @@ struct NewStoryView: View {
 
     private func createStory() {
         guard areAllFieldsFilled else { return }
-        self.createdStory = Story(name: name, ticketReference: ticketID, epic: epic)
+        let newStory = Story(name: name, ticketReference: ticketID, epic: epic)
+        self.createdStory = newStory
+        if var sprintStories = Store.data[sprint] {
+            sprintStories.append(newStory)
+            Store.data[sprint] = sprintStories
```
Comment (zarghol) [inline RunningOrder/Story/NewStoryView.swift:50]: shouldn't be here

File: RunningOrder/Sprint/SprintDetail.swift (hunk 2 above)
Comment (lucas-barbero) [reply RunningOrder/Sprint/SprintDetail.swift:13]: Since we can't edit a sprint for now we can just use a let

File: RunningOrder/Sprint/SprintDetail.swift (hunk 3 above)
Comment (lucas-barbero) [reply RunningOrder/Sprint/SprintDetail.swift:25]: This one is triggered before the  NavigationView one, isn't it ?

File: RunningOrder/Sprint/SprintList.swift (hunk 5 above)
Comment (lucas-barbero) [reply RunningOrder/Sprint/SprintList.swift:58]: Because we design the SprintList like a Sidebar, I think it's more convenient to put the style here and not in the root view which use the list
